nautobot/extras/signals.py

```diff
@@ -455,6 +456,13 @@ def job_result_delete_associated_files(instance, **kwargs):
             file_proxy.file.delete()
 
 
+@receiver(post_save, sender=JobModel)
+def add_default_job_queue_to_job_queues(instance, **kwargs):
+    # Add specified default_job_queue to job.job_queues if it is not included.
+    if not instance.job_queues.filter(pk=instance.default_job_queue.pk).exists():
+        instance.job_queues.add(instance.default_job_queue)
```

Comment:
This still feels like the wrong place to do this. In response to your question about whether doing this in the `m2m_changed` signal will be problematic, StackOverflow at least [suggests](https://stackoverflow.com/questions/26493254/using-djangos-m2m-changed-to-modify-what-is-being-saved-pre-add) that updating `pk_set` in the signal handler is a valid way to influence the set of M2M relations.